packages/perseus-editor/src/widgets/free-response-editor.tsx

```diff
@@ -108,6 +121,32 @@ class FreeResponseEditor extends React.Component<Props> {
                         }
                     />
                 </label>
+                <label
+                    className={css(styles.textOptionWithLabelContainer)}
+                    htmlFor="allow-unlimited-characters"
+                >
+                    <HeadingSmall>Allow unlimited characters</HeadingSmall>
+                    <Checkbox
+                        checked={this.props.allowUnlimitedCharacters}
+                        id="allow-unlimited-characters"
+                        onChange={(val) =>
+                            this.props.onChange({
+                                allowUnlimitedCharacters: val,
```

Comment:
Might it make sense to update the `characterLimit` when `allowUnlimitedCharacters` changes? I could imagine setting it to `0` when `allowUnlimitedCharacters === true`, and setting it to the default when `allowUnlimitedCharacters === false`.

Reply:
This is kind of related to my suggestion of merging `allowUnlimitedCharacters` and `characterLimit` so that `characterLimit === null` means "allow unlimited characters"; it wouldn't allow for a state where `characterLimit` looks like it's being used but it's actually not.

Reply:
Personally, I prefer a dedicated `allowUnlimitedCharacters` variable, for two reasons:
1. It's more explicit. If a developer looks at the `characterLimit` and sees it's `null`, does that mean it's just not set (maybe accidentally) or does it have a valid meaning?
2. It maps more clearly onto the editing UI. We want the user to have to separately choose yes or no for unlimited characters, and the data reflects that.

I'd be open to allowing `null` for the `characterLimit` and nulling it out when the content editor selects `allowUnlimitedCharacters === false`, but I'd prefer to keep the two values separate.